backend/src/controller/chatController.ts

```diff
@@ -169,15 +169,13 @@ async function handleHigherLevelChat(
 	}
 	const combinedDefenceReport = combineChatDefenceReports(defenceReports);
 
-	if (combinedDefenceReport.isBlocked) {
-		// restore the original chat history
-		// add user message to the chat history (not as completion)
-		updatedChatHistory = pushMessageToHistory(chatHistory, {
-			completion: null,
-			chatMessageType: CHAT_MESSAGE_TYPE.USER,
-			infoMessage: message,
-		});
-	}
+	const updatedChatHistory = combinedDefenceReport.isBlocked
```

Comment:
I liked that there was a code comment here originally, explaining that we are using the original chat history if this chat was blocked, as that is easily missed.